Forms/DrawForm/WeightTable.cs

```diff
@@ -0,0 +1,125 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Windows.Forms;
+using GraphModelDraw;
+
+namespace Forms.DrawForm
+{
+    class WeightTable : Panel
+    {
+
+        public WeightTable(int width, int height, int positionX, int positionY)
+        {
+            Size = new System.Drawing.Size(width, height);
+
+            Location = new System.Drawing.Point(positionX, positionY);
+
+            Dock = DockStyle.Right;
+
+            AutoScroll = true;
+
+            BorderStyle = BorderStyle.Fixed3D;
+
+        }
+  
+    }
+
+
+     class MatrixWeightPanel
+    {
+
+        private WeightTable weightTable;
+
+        private List<List<InputCountBox>> matrix;
+
+        public MatrixWeightPanel(WeightTable weightTable, List<List<InputCountBox>> matrix)
+        {
+
+            this.weightTable = weightTable;
+            this.matrix = matrix;
+
+        }
+
+        public void DrawingMatrix(int rows)
```

Comment:
Why do you pass a parameter here, if you have a "matrix" field that already contains information about the number of elements? This is not safe.